src/MediaUploader.php

```diff
@@ -438,12 +443,34 @@ public function upload()
 
         $this->verifyDestination($model);
 
+        if (is_callable($this->before_save)) {
+            call_user_func($this->before_save, $model, $this->source);
+        }
+
         $this->filesystem->disk($model->disk)->put($model->getDiskPath(), $this->source->contents());
         $model->save();
 
         return $model;
     }
 
+    public function modifySource(callable $callable)
+    {
+        $newSource = call_user_func($callable, $this->source);
+
+        return static::fromSource($newSource);
+    }
```

Comment:
I still don't think that this is necessary. Why not just modify the file yourself before passing it to the `fromSource()` method?

Reply:
I'm overthinking, you're absolutely right.